packages/records/lib/models/records.ts

```diff
@@ -200,12 +200,14 @@ export async function upsert({
     connectionId,
     environmentId,
     model,
+    syncJobId,
     softDelete = false
 }: {
     records: FormattedRecord[];
     connectionId: number;
     environmentId: number;
     model: string;
+    syncJobId: number;
```

Comment:
One last comment, we create a sync job for each webhook. ConnectionId or syncId might be better suited maybe

Reply:
the original goal was to prevent race condition from the same job execution but you are right that webhooks can affect the same dataset. 
The challenge of webhook concurrency is a whole other topic but since we are trying to prevent race condition let's have a lock on connection/model. I'll change that on Tuesday